src/foraging_gui/Foraging.py

```diff
@@ -3963,6 +3959,23 @@ def _StartTrialLoop(self,GeneratedTrials,worker1,worker_save):
                 #generate a new trial
                 if self.NewTrialRewardOrder==1:
                     GeneratedTrials._GenerateATrial(self.Channel4)   
+                '''
+                ### Save data in the main thread. Keep it temporarily for testing ### 
```

Comment:
Did you mean to remove this part, saving data in the main thread?

Reply:
Yes, I discarded this the strategy to 'save data in the main thread'. Instead, I moved the previous `saving code` to the after of `GeneratedTrials._GenerateATrial(self.Channel4)` to make sure they are not running simultaneously.

Reply:
I'm confused. Line 3967 shows saving the data in the main thread, then line 3975 shows saving in a separate thread. Did you intend to remove the saving in the main thread?

Reply:
The code for "saving data in the main thread" has been commented out with ''' '''

Reply:
Oh duh. I didn't see the multiline comment ```, I was looking for a # before each line